jax/_src/api_util.py

```diff
@@ -322,6 +322,11 @@ def shaped_abstractify(x):
   except TypeError:
     pass
 
+  try:
```

Comment:
Why not use isinstance here and above instead of try;except?

Reply:
Just force of habit from years of Python duck-typing 😁

Reply:
Can you change it to use isinstance? It'll be faster I think!

Without needing to execute and get an error, you can check up-front and then execute (that in my mind seems faster but we can measure that).

Reply:
Sure I'll give that a try